bundles/org.eclipse.text.quicksearch/src/org/eclipse/text/quicksearch/internal/ui/QuickSearchDialog.java

```diff
@@ -531,9 +547,10 @@ public boolean close() {
 	 *           settings used to store dialog
 	 */
 	protected void storeDialog(IDialogSettings settings) {
-		String currentSearch = pattern.getText();
-		settings.put(DIALOG_LAST_QUERY, currentSearch);
-		settings.put(DIALOG_PATH_FILTER, searchIn.getText());
+		DialogSettings dialogSettings = (DialogSettings) settings;
+		dialogSettings.inputNewSearchFilter(searchIn.getText());
```

Comment:
Since the casted variable is only used once, I think it can be inlined:
```suggestion
		((DialogSettings)dialogSettings).inputNewSearchFilter(searchIn.getText());
```

Reply:
done